src/openrct2/object/ObjectManager.cpp

```diff
@@ -69,6 +69,12 @@ class ObjectManager final : public IObjectManager
 
     Object* GetLoadedObject(ObjectType objectType, size_t index) override
     {
+        // This is sometimes done deliberately (to avoid boilerplate), so no need to log_warn for this.
+        if (index == OBJECT_ENTRY_INDEX_NULL)
```

Comment:
Wouldn't it be better to solve this a bit higher up? We don't need to call this if we know that this will result null anyway.

Reply:
It would mean extra boilerplate in a few dozen places, harming readability. We already get this kind of spam when working with music objects. That’s why I preferred this solution.

Reply:
Fair enough.

Reply:
This is fine, it means anything can freely pass an object ID into GetLoadObject without worrying about the range as it then only has to check `nullptr` afterwards. Not doing this means you have to make everything that calls this check for invalid range and you risk some callers not doing that.

Reply:
Well it is already protected against values out of the valid range, I just think it would make sense to check this in lets say GetSurfaceObject before calling GetContext(), GetObjectManager(), GetObject etc but only for the sake of avoiding those calls but it doesn't matter too much.

Reply:
That makes senses, although there will still be some things that will call GetObject directly and not be in a wrapper like GetSurfaceObject, so you'd still need both.